source/validate_instruction.cpp

```diff
@@ -139,14 +154,33 @@ spv_result_t CapCheck(ValidationState_t& _,
     } else {
       // Check the operand word as a whole.
       const auto caps = RequiredCapabilities(_, operand.type, word);
-      if (!_.HasAnyOf(caps)) {
+      if (!_.HasAnyOfCapabilities(caps)) {
         return CapabilityError(_, i + 1, opcode, ToString(caps, _.grammar()));
       }
     }
   }
   return SPV_SUCCESS;
 }
 
+// Checks that all required extensions were declared in the module.
+spv_result_t ExtensionCheck(ValidationState_t& _,
+                            const spv_parsed_instruction_t* inst) {
+  const SpvOp opcode = static_cast<SpvOp>(inst->opcode);
+  for (int i = 0; i < inst->num_operands; ++i) {
+    const auto& operand = inst->operands[i];
+    const uint32_t word = inst->words[operand.offset];
+    const ExtensionSet required_extensions =
+        RequiredExtensions(_, operand.type, word);
+    if (!_.HasAnyOfExtensions(required_extensions)) {
+      return _.diag(SPV_ERROR_MISSING_EXTENSION)
+          << "Operand " << word << " of " << spvOpcodeString(opcode)
```

Comment:
This could be ambiguous, in theory.  The word value  might be valid in another operand position.  At the very least you should print the operand index as well.